### PR TITLE
Add a 'preferred-modules' switch checking for possible module replacement

### PR DESCRIPTION
**Pull request by PaulRenvoise**
## Steps

- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description

This PR adds a new checker and a new option to `ImportsChecker`: `preferred-module` and `--preferred-modules`. This checker allows users to specify a comma-separated list of module:preferred-module that will make pylint suggest a replacement when a non-preferred module import is detected.

For example if a code base uses `ujson` instead of `json` for whatever reason, it can be specified in pylint as: `preferred-modules='json:ujson'`.

The default value for preferred-modules is `xml:defusedxml,lxml:defusedxml' based on Bandit's blacklist (https://docs.openstack.org/bandit/latest/api/bandit.blacklists.html#b313-b320-xml), but this can be changed.

## Type of Changes
|   | Type |
| ------------- | ------------- |
| ✓  | :sparkles: New feature |

### CODE REVIEW

File: pylint/checkers/imports.py [hunk 1]
```diff
@@ -277,6 +283,7 @@ def _make_graph(filename, dep_info, sect, gtype):
 
 DEFAULT_STANDARD_LIBRARY = ()
 DEFAULT_KNOWN_THIRD_PARTY = ("enchant",)
+DEFAULT_PREFERRED_MODULES = ("xml:defusedxml", "lxml:defusedxml")
```
Comment (PCManticore) [inline pylint/checkers/imports.py]: I don't think we should be recommending anything at this point and should be let at the latitude of each project using `pylint` to define their own set of preferred modules.

Comment (PaulRenvoise) [reply pylint/checkers/imports.py]: Sure thing, I changed that in 2dc8e3a

**Comment by PCManticore** [comment]
@PaulRenvoise There seems to be a broken merge, can you remove the commits that are not from this branch?

**Comment by PCManticore** [comment]
Hi @PaulRenvoise I merged this PR manually. Thank you so much for sending a PR!

**Comment by PaulRenvoise** [comment]
Hi @PCManticore,
Sorry for not being reactive the past few days, and thanks a lot for accepting and merging this PR! :)

**Pull request closed without merging**